PhysicsTools/PatAlgos/python/jetCollectionTools.py

```diff
@@ -0,0 +1,621 @@
+  chHEF     = jetTable.variables.chHEF,
+  neHEF     = jetTable.variables.neHEF,
+  chEmEF    = jetTable.variables.chEmEF,
+  neEmEF    = jetTable.variables.neEmEF,
+  muEF      = jetTable.variables.muEF,
+  rawFactor = jetTable.variables.rawFactor,
+  jetId     = jetTable.variables.jetId,
+  jercCHPUF = jetTable.variables.jercCHPUF,
+  jercCHF   = jetTable.variables.jercCHF,
+)
+
+
+#============================================
+#
+# JetInfo
+#
+#============================================
+class GenJetInfo(object):
+  def __init__(self, jet, inputCollection):
+    self.jet = jet
+    self.inputCollection = inputCollection
+    algoKey     = 'algo'
+    sizeKey     = 'size'
+    recoKey     = 'reco'
+    jetRegex = re.compile(
+      r'(?P<{algo}>({algoList}))(?P<{size}>[0-9]+)gen'.format(
+        algo     = algoKey,
+        algoList = '|'.join(supportedJetAlgos.keys()),
+        size     = sizeKey,
+      )
+    )
+
+    jetMatch = jetRegex.match(jet.lower())
+    if not jetMatch:
+      raise RuntimeError('Invalid jet collection: %s' % jet)
+    self.jetAlgo     = jetMatch.group(algoKey)
+    self.jetSize     = jetMatch.group(sizeKey)
+    self.jetSizeNr = float(self.jetSize) / 10.
+
+#============================================
+#
+# GenJetAdder
+#
+#============================================
+class GenJetAdder(object):
+  def __init__(self):
+    self.prerequisites = []
+    self.main = []
+    self.gpLabel = "prunedGenParticles"
+
+  def getSequence(self, proc):
+    tasks = self.prerequisites + self.main
+
+    resultSequence = cms.Sequence()
+    for idx, task in enumerate(tasks):
+      if idx == 0:
+        resultSequence = cms.Sequence(getattr(proc, task))
+      else:
+        resultSequence.insert(idx, getattr(proc, task))
+    return resultSequence
+  
+  def addGenJetCollection(self,
+      proc,
+      jet,
+      name,
+      doc,
+      inputCollection    = "",
+      genName            = "",
+      minPt              = 5.,
+    ):
+    print("custom_jme_cff::GenJetAdder::addGenJetCollection: Adding Gen Jet Collection: {}".format(jet))
+    currentTasks = []
+    #
+    # Decide which jet collection we're dealing with
+    #
+    jetLower = jet.lower()
+    jetUpper = jet.upper()
+    tagName  = jetUpper
+    genJetInfo = GenJetInfo(jet,inputCollection)
+
+    #
+    # Skip AK4GenJets and AK8GenJets.
+    # They're already available in the default NanoAOD
+    #
+    if genJetInfo.jetSize  == "4" and genJetInfo.jetAlgo == "ak":
+      pass
+    elif genJetInfo.jetSize  == "8" and genJetInfo.jetAlgo == "ak":
+      pass
+    #=======================================================
+    #
+    # If jet collection in MiniAOD is  not 
+    # specified, build the jet collection.
+    #
+    #========================================================
+    if not inputCollection:
+      print("custom_jme_cff::GenJetAdder::addGenJetCollection: inputCollection in NanoAOD not specified. Building genjet collection now")
```

Comment:
the comments "in NanoAOD" are not really appropriate if we want this code in  PhysicsTools/PatAlgos.
As is, there is no strong coupling to nano